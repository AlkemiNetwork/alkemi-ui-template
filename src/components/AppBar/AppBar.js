import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import MuiAppBar from "@material-ui/core/AppBar";
import { withStyles } from "@material-ui/core/styles";

import AppBarToolbar from "./Toolbar";

const styles = theme => ({
    appBar: {
        position: "fixed",
        top: 0,
        zIndex: theme.zIndex.drawer + 1,
        transition: theme.transitions.create(["width", "margin"], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.leavingScreen
        })
    },
    appBarShifted: {
        marginLeft: 240,
        width: `calc(100% - ${240}px)`,
        transition: theme.transitions.create(["width", "margin"], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.enteringScreen
        }),
        [theme.breakpoints.down("sm")]: {
            width: "100%"
        }
    }
});

const AppBar = ({ classes, shifted }) => {
    return (
        <MuiAppBar
            position="absolute"
            className={classNames(
                classes.appBar,
                shifted && classes.appBarShifted
            )}
        >
            <AppBarToolbar shifted={shifted} />
        </MuiAppBar>
    );
};

AppBar.propTypes = {
    classes: PropTypes.object.isRequired,
    shifted: PropTypes.bool
};

export default withStyles(styles)(AppBar);
