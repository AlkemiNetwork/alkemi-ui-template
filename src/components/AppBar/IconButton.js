import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { withStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";

const styles = theme => ({
    menuButton: {
        marginLeft: 12,
        marginRight: 36
    },
    hide: {
        [theme.breakpoints.up("sm")]: {
            display: "none"
        }
    }
});

const AppbarButton = ({ classes, onClick, shifted }) => (
    <IconButton
        color="inherit"
        aria-label="open drawer"
        onClick={onClick}
        className={classNames(classes.menuButton, shifted && classes.hide)}
    >
        <MenuIcon />
    </IconButton>
);

AppbarButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    shifted: PropTypes.bool
};

export default withStyles(styles)(AppbarButton);
