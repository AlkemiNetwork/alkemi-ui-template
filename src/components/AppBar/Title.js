import React from "react";
import PropTypes from "prop-types";

import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from "@material-ui/core/Typography";

const AppBarTitle = ({ onClick }) => (
    <ButtonBase onClick={onClick}>
        <Typography variant="title" color="inherit" noWrap>
            ConsenSys React Boilerplate
        </Typography>
    </ButtonBase>
);

AppBarTitle.propTypes = {
    onClick: PropTypes.func.isRequired
};

export default AppBarTitle;
