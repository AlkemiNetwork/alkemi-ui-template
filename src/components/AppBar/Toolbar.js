import React from "react";
import PropTypes from "prop-types";

import Toolbar from "@material-ui/core/Toolbar";
import withWidth, { isWidthUp } from "@material-ui/core/withWidth";

import AppBarTitle from "../../containers/AppBar/Title";
import AppBarIconButton from "../../containers/AppBar/IconButton";

const AppBarToolbar = ({ width, shifted }) => (
    <Toolbar disableGutters={!shifted || isWidthUp("md", width)}>
        <AppBarIconButton />
        <AppBarTitle />
    </Toolbar>
);

AppBarToolbar.propTypes = {
    width: PropTypes.string.isRequired,
    shifted: PropTypes.bool
};

export default withWidth()(AppBarToolbar);
