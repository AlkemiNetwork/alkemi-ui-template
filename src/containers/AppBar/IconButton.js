import { connect } from "react-redux";

import { openSidebar } from "../../redux/actions/ui";
import AppBarIconButton from "../../components/AppBar/IconButton";

const mapStateToProps = state => ({
    shifted: state.ui.sidebar.open
});

const mapDispatchToProps = {
    onClick: openSidebar
};

export default connect(mapStateToProps, mapDispatchToProps)(AppBarIconButton);
