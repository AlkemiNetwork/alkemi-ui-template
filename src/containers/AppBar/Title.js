import { connect } from "react-redux";
import { push } from "connected-react-router";

import AppBarTitle from "../../components/AppBar/Title";
import { HOME } from "../../constants/routes";

const mapDispatchToProps = {
    onClick: () => push(HOME)
};

export default connect(undefined, mapDispatchToProps)(AppBarTitle);
