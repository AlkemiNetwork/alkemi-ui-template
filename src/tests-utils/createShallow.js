import { shallow as _shallow } from "enzyme";

const createShallow = ({ dive }) => {
    const shallow = (node, options) => {
        let wrapper = _shallow(node, options);

        for (let i = 0; i < dive; i++) {
            if (wrapper.name().match(/^WithWidth(.+)$/)) {
                // Diving through a Meterial-UI WidthWith component can not be performed
                // by default because of React.Fragment used in withWidth
                // It is a known issue: https://github.com/airbnb/enzyme/issues/1213
                wrapper = wrapper
                    .dive()
                    .children()
                    .at(0);
            } else {
                wrapper = wrapper.dive();
            }
        }
        return wrapper;
    };

    return shallow;
};

export default createShallow;
