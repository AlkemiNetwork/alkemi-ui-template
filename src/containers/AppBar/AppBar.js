import { connect } from "react-redux";

import AppBar from "../../components/AppBar/AppBar";

const mapStateToProps = state => ({
    shifted: state.ui.sidebar.open
});

export default connect(mapStateToProps)(AppBar);
