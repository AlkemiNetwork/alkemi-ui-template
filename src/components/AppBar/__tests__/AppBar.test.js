import React from "react";
import toJson from "enzyme-to-json";

import AppBar from "../AppBar";
import AppBarToolbar from "../Toolbar";
import createShallow from "../../../tests-utils/createShallow";

describe("<AppBar />", () => {
    let shallow;
    let wrapper;

    beforeAll(() => {
        shallow = createShallow({ dive: 2 });
    });

    beforeEach(() => {
        wrapper = shallow(<AppBar />);
    });

    it("has className appBar", () => {
        expect(wrapper.props().className).toContain("appBar");
    });

    it("renders 1 <AppBarToolbar /> components", () => {
        expect(wrapper.find(AppBarToolbar).length).toEqual(1);
    });

    describe("when shifted=true", () => {
        beforeEach(() => {
            wrapper = shallow(<AppBar shifted />);
        });

        it("matches snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("has className appBarShifted", () => {
            expect(wrapper.props().className).toContain("appBarShifted");
        });
    });

    describe("when shifted=false", () => {
        beforeEach(() => {
            wrapper = shallow(<AppBar />);
        });

        it("matches snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("does not have className appBarShift", () => {
            expect(wrapper.props().className).not.toContain("appBarShift");
        });
    });
});
