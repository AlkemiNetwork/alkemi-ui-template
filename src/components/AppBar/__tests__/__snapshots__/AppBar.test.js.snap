// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBar /> when shifted=false matches snapshot 1`] = `
<AppBar
  className="AppBar-appBar-1"
  classes={
    Object {
      "colorDefault": "MuiAppBar-colorDefault-9",
      "colorPrimary": "MuiAppBar-colorPrimary-10",
      "colorSecondary": "MuiAppBar-colorSecondary-11",
      "positionAbsolute": "MuiAppBar-positionAbsolute-5",
      "positionFixed": "MuiAppBar-positionFixed-4",
      "positionRelative": "MuiAppBar-positionRelative-8",
      "positionStatic": "MuiAppBar-positionStatic-7",
      "positionSticky": "MuiAppBar-positionSticky-6",
      "root": "MuiAppBar-root-3",
    }
  }
  color="primary"
  position="absolute"
>
  <WithTheme(WithWidth(AppBarToolbar)) />
</AppBar>
`;

exports[`<AppBar /> when shifted=true matches snapshot 1`] = `
<AppBar
  className="AppBar-appBar-1 AppBar-appBarShifted-2"
  classes={
    Object {
      "colorDefault": "MuiAppBar-colorDefault-9",
      "colorPrimary": "MuiAppBar-colorPrimary-10",
      "colorSecondary": "MuiAppBar-colorSecondary-11",
      "positionAbsolute": "MuiAppBar-positionAbsolute-5",
      "positionFixed": "MuiAppBar-positionFixed-4",
      "positionRelative": "MuiAppBar-positionRelative-8",
      "positionStatic": "MuiAppBar-positionStatic-7",
      "positionSticky": "MuiAppBar-positionSticky-6",
      "root": "MuiAppBar-root-3",
    }
  }
  color="primary"
  position="absolute"
>
  <WithTheme(WithWidth(AppBarToolbar))
    shifted={true}
  />
</AppBar>
`;
